tidy3d/plugins/mode/mode_solver.py

```diff
@@ -579,6 +579,104 @@ def to_mode_solver_monitor(self, name: str) -> ModeSolverMonitor:
             name=name,
         )
 
+    def add_source(
+        self,
+        source_time: SourceTime,
+        direction: Direction,
+        mode_index: pydantic.NonNegativeInt = 0,
+        ) -> Simulation:
+
+        """Creates :class:`Simulation` from a :class:`ModeSolver` instance plus additional
+        specifications.
+
+        Parameters
+        ----------
+        source_time: :class:`.SourceTime`
+            Specification of the source time-dependence.
+        direction : Direction
+            Whether source will inject in ``"+"`` or ``"-"`` direction relative to plane normal.
+        mode_index : int = 0
+            Index into the list of modes returned by mode solver to use in source.
+
+        Returns
+        -------
+        :class:`.Simulation`
+            Copy of the simulation with a mode source with specifications taken
+            from the ModeSolver instance and the method inputs.
+        """
+
+        mode_source = self.to_source(mode_index=mode_index, direction=direction, source_time=source_time)
+        new_sources = self.simulation.sources
+        if not new_sources:
+            new_sources = [mode_source]
+        else:
+            new_sources = new_sources + tuple([mode_source])
+        newSim = self.simulation.copy(update=dict(sources=new_sources))
```

Comment:
in python, the convention we use (from [PEP8](https://peps.python.org/pep-0008/#method-names-and-instance-variables)) is to use lowercase + underscores, eg "`new_sim`" instead of camel case "`newSim`".